api/dataset_api.go

```diff
@@ -32,59 +33,59 @@ func (api *DatasetAPI) GetInstances(ctx context.Context, vars url.Values) (insta
 
 // SetImportObservationTaskComplete marks the import observation task state as completed for an instance
 func (api *DatasetAPI) SetImportObservationTaskComplete(ctx context.Context, instanceID string) (isFatal bool, err error) {
-	err = api.Client.PutInstanceImportTasks(ctx, api.ServiceAuthToken, instanceID,
+	_, err = api.Client.PutInstanceImportTasks(ctx, api.ServiceAuthToken, instanceID,
 		dataset.InstanceImportTasks{
 			ImportObservations: &dataset.ImportObservationsTask{
 				State: dataset.StateCompleted.String(),
 			},
-		},
+		}, headers.IfMatchAnyETag,
 	)
 	isFatal = errorChecker(ctx, "SetImportObservationTaskComplete", err, &log.Data{})
 	return
 }
 
 // UpdateInstanceWithNewInserts increments the observation inserted count for an instance
 func (api *DatasetAPI) UpdateInstanceWithNewInserts(ctx context.Context, instanceID string, observationsInserted int32) (isFatal bool, err error) {
-	err = api.Client.UpdateInstanceWithNewInserts(ctx, api.ServiceAuthToken, instanceID, observationsInserted)
+	_, err = api.Client.UpdateInstanceWithNewInserts(ctx, api.ServiceAuthToken, instanceID, observationsInserted, headers.IfMatchAnyETag)
 	isFatal = errorChecker(ctx, "UpdateInstanceWithNewInserts", err, &log.Data{})
 	return
 }
 
 // UpdateInstanceWithHierarchyBuilt marks a hierarchy build task state as completed for an instance.
 func (api *DatasetAPI) UpdateInstanceWithHierarchyBuilt(ctx context.Context, instanceID, dimensionID string) (isFatal bool, err error) {
-	err = api.Client.PutInstanceImportTasks(ctx, api.ServiceAuthToken, instanceID,
+	_, err = api.Client.PutInstanceImportTasks(ctx, api.ServiceAuthToken, instanceID,
```

Comment:
I know you weren't the one who wrote it initially but adding the headers.IfMatchanyEtag bought it to my attention.. the formatting is slightly off with these functions. Recommend:
```suggestion
	_, err = api.Client.PutInstanceImportTasks(
		ctx,
		api.ServiceAuthToken,
		instanceID,
		dataset.InstanceImportTasks{
			BuildSearchIndexTasks: []*dataset.BuildSearchIndexTask{
				&dataset.BuildSearchIndexTask{
					DimensionName: dimensionID,
					State: dataset.StateCompleted.String(),
				},
			},
			headers.IfMatchAnyETag,
		},
	)
```